repository/src/main/java/org/apache/atlas/discovery/EntityLineageService.java

```diff
@@ -198,6 +203,33 @@ public AtlasLineageInfo getAtlasLineageInfo(String guid, LineageOnDemandRequest
         return ret;
     }
 
+    private void handleDefaultParams(LineageOnDemandRequest lineageOnDemandRequest) {
+        for (Map.Entry<String, LineageOnDemandConstraints> constraintEntry : lineageOnDemandRequest.getConstraints().entrySet()) {
+            if (constraintEntry.getValue().getInputRelationsLimit() == 0) {
```

Comment:
What is the user want it to be actually 0?

Reply:
I think check only for null and move the logic to getAndValidateLineageConstraintsByGuid

Reply:
discussed

Reply:
addressed